atc/db/team.go

```diff
@@ -163,6 +164,18 @@ func (t *team) FindWorkerForVolume(handle string) (Worker, bool, error) {
 	}))
 }
 
+func (t *team) FindWorkersForResourceCache(rcId int) ([]Worker, error) {
+	return getWorkers(
+		t.conn, workersQuery.
+			Join("volumes v ON v.worker_name = w.name").
+			Join("worker_resource_caches wrc ON v.worker_resource_cache_id = wrc.id").
+			Join("resource_caches rc ON rc.id = wrc.resource_cache_id").
+			Where(sq.And{
```

Comment:
We probably want to only fetch from a worker that the team has access to. So if there are team workers that are not part of this team or tagged workers that have the resource cache on them, we don't want to use that worker.

Reply:
I guess not really. I think "use that worker" means "run containers on the worker". But resource cache is completely distinguished by `source`, where `source` may include URL and credential to access the resource, it's very possible that multiple teams may share a common resource, like a docker image. So I feel `team` and `tag` don't really play significant role with resource cache.

Reply:
Right, but I think we still don't want to use a team worker unless the get step was run from that team or a tagged worker if the step itself is tagged for that worker. For example, there can be the case where a team worker has access to a private network that other team do not have access to. They would expect that only their team would have access to that private network but if we shared volumes between team workers then that would not hold true. The resource cache being distinguished by source and type does not completely guarantee that it will always be unique (ex. a repo in two private networks can have contain the same items).

Reply:
That makes sense, though I would not be true in our deployments. I'll update the SQL.

Reply:
@clarafu fixed, please take a look.

Reply:
I think we also don't want to use a cache on a tagged worker if the get step does not have a matching tag!